cmd/registry/cmd/upload/artifact.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upload
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/apigee/registry/cmd/registry/controller"
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/rpc"
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+func artifactCommand(ctx context.Context) *cobra.Command {
+	var parent string
+	cmd := &cobra.Command{
+		Use:   "artifact FILE_PATH --parent=value",
+		Short: "Upload an artifact",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			artifactPath := args[0]
+			if artifactPath == "" {
```

Comment:
NIT: Suggest to rename this as `artifactFilePath`, my first interpretation of `artifactPath` was path in the registry like  `projects/demo/locations/global/artifacts/manifest`

Reply:
Thanks - I've changed the name